### PR TITLE
CY-3381 Implement the external db case

### PR DESCRIPTION
**Pull request by YoniItzhak**

### CODE REVIEW

**Comment by YoniItzhak** [comment]
This PR implements the external-db option. The implementation handles both the 6 VMs case (3 Manager + 3 RabbitMQ's) and the 3 nodes case.

File: cfy_cluster_setup/config_files_templates/manager_config.yaml [hunk 1]
```diff
@@ -21,17 +21,24 @@ rabbitmq:
...
+  {% if external_db_configuration %}
+  ca_path: {{ external_db_configuration.ca_path }}
+  {% else %}
+  ca_path: {{ ca_path }}
+  ssl_enabled: true
+  postgres_password: {{ creds.postgresql.postgres_password }}
+  {% endif %}
 
 postgresql_client:
+  {% if external_db_configuration %}
```
Comment (geokala) [inline cfy_cluster_setup/config_files_templates/manager_config.yaml:35]: I think you might want some minuses (-%} or {%-) to stop getting ugly extra whitespace with this.

File: cfy_cluster_setup/main.py [hunk 2]
```diff
@@ -494,6 +562,12 @@ def main():
         default=False,
         help='Using a three nodes cluster.')
 
+    generate_config_args.add_argument(
+        '--external-db',
+        action='store_true',
```
Comment (geokala) [inline cfy_cluster_setup/main.py:567]: Should this maybe want an address (IP or hostname) instead of just being a flag?

Comment (YoniItzhak) [reply cfy_cluster_setup/main.py:567]: It wants a lot more than that, unfortunately. This is why I added a [whole section for this use case in the configuration file](https://github.com/cloudify-cosmo/cloudify-cluster-setup/blob/e55cadd89cf2cb4c5d24f022e1c442dcfdb31a9e/cfy_cluster_setup/cfy_cluster_config_files/cfy_external_db_cluster_config.yaml#L65-L86).

Comment (geokala) [reply cfy_cluster_setup/main.py:567]: Oh, that sucks. Fair enough then, let's go with this route for now and see if we can do anything better with some background cfy_manager improvements in future.

File: cfy_cluster_setup/config_files_templates/manager_config.yaml (hunk 1 above)
Comment (YoniItzhak) [reply cfy_cluster_setup/config_files_templates/manager_config.yaml:35]: I don't think I fully understand this comment. Isn't this the Jinja template syntax `{% parameter %}`?

File: cfy_cluster_setup/main.py (hunk 2 above)
Comment (YoniItzhak) [reply cfy_cluster_setup/main.py:567]: Sounds good.

File: cfy_cluster_setup/config_files_templates/manager_config.yaml (hunk 1 above)
Comment (geokala) [reply cfy_cluster_setup/config_files_templates/manager_config.yaml:35]: Sure, but you can also add - next to the % signs to suppress some whitespace. As written you will get something like:
somestuff

postgresql_client:
  
  say hello
  

otherstuff

Comment (geokala) [reply cfy_cluster_setup/config_files_templates/manager_config.yaml:35]: ...though with indentation. Thanks, github.

Comment (geokala) [reply cfy_cluster_setup/config_files_templates/manager_config.yaml:35]: With a -%} on the first if you will get something like:
somestuff

postgresql_client:
  say hello
  

otherstuff

Comment (geokala) [reply cfy_cluster_setup/config_files_templates/manager_config.yaml:35]: Github's ui has also eaten the extra newlines
I really detest this editor...

Comment (geokala) [reply cfy_cluster_setup/config_files_templates/manager_config.yaml:35]: Anyway, I suggest trying rendering it by hand to try to get to the point that you don't have:
`somestuff

postgresql_client:
  say hello
  

otherstuff`

Comment (geokala) [reply cfy_cluster_setup/config_files_templates/manager_config.yaml:35]: ...even when I use the insert code feature? Really?
Ok, so I suggest rendering it by hand to try to get to the point that it looks nice and cleanly written.

Comment (YoniItzhak) [reply cfy_cluster_setup/config_files_templates/manager_config.yaml:35]: Yep, I tried it now on a dummy example and the result was that all the blank lines were deleted.
 
```
rabbitmq:
  ca_path:
  cluster_members:

{%-if '1' in node.hostname-%}
networks:
  load_balancer: {{ node.hostname }}
{%-endif-%}

postgresql_server:
  ssl_enabled: true
  ca_path:
```

turned out to: 
```
rabbitmq:
  ca_path:
  cluster_members:networks:
  load_balancer: yoni-1postgresql_server:
  ssl_enabled: true
  ca_path:
```

Comment (geokala) [reply cfy_cluster_setup/config_files_templates/manager_config.yaml:35]: Looks good!

**Review by geokala — APPROVED** [review]

**Pull request merged**